src/backend/commands/tablecmds.c

```diff
@@ -8175,6 +8187,19 @@ ATExecSetNotNull(AlteredTableInfo *tab, Relation rel,
 	AttrNumber	attnum;
 	Relation	attr_rel;
 
+	if (RelationIsExternal(rel))
+	{
+		ExtTableEntry *ext_table_entry = GetExtTableEntry(RelationGetRelid(rel));
+
+		if (!ext_table_entry->iswritable)
+		{
+			ereport(WARNING,
```

Comment:
Shall it be WARNING or ERROR here? WARNING means operation completed but may be not in the expected way. ERROR means operation was not performed at all, which is the case for SetNotNull.

Reply:
It could be an error, however the general consensus was that we are allowing all common NOT NULL operations while ignoring the actual constraint. That way the behavior is consistent with NOT NULL specified in the table definition.